avro-fastserde/src/main/java/com/linkedin/avro/fastserde/ByteBufferBackedPrimitiveFloatList.java

```diff
@@ -185,8 +185,11 @@ public float getPrimitive(int i) {
     if (i >= size) {
       throw new IndexOutOfBoundsException("Index " + i + " out of bounds.");
     }
-    cacheFromByteBuffer();
-    return elements[i];
+
```

Comment:
this change actually degrades the performance of the "common" case?
if some user wants to be "more efficient" by writing a for loop that uses getPrimitive(i) their performance actually suffers because they would decode one element at a time instead of hitting cacheFromByteBuffer()

Reply:
actually it does not. If user accesses same element multiple times then it might have some impact, as it will deserialize the same element multiple times. but for iterating once over all the elements, the amortized cost would be similar.

Reply:
yeah, I have the similar concern.
What is the scenario that user will only want to access some of the float array?
Do you have the benchmark result for accessing all the elements w/o this change?

Reply:
I think there are some teams doing sparse access of the array, ie accessing only few or every n-th element of the array.
Also accessing every element does give better perf. Ideally though both should have same amortized cost. 

without the change: (~107 ns per op)
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess avgt 3 107.528 ± 6.851 ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess:·gc.count avgt 3 112.000 counts
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess:·gc.time avgt 3 120.000 ms


with the change: (81 ns per op)
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess avgt 3 81.539 ± 1.648 ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess:·gc.count avgt 3 106.000 counts
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess:·gc.time avgt 3 109.000 ms

Reply:
results with running more iteration(1M) and larger array size (1k)

without change:
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess                              avgt    3   4360.619 ± 2842.226   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess8                            avgt    3   1692.187 ± 1270.572   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess16                           avgt    3   1546.228 ±  181.865   ns/op

with change:
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess                                 avgt    3   4741.339 ±  754.452   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess8                               avgt    3    636.909 ±  265.902   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess16                             avgt    3    381.874 ±  321.521   ns/op

Reply:
So it is slightly worse with the change?

Reply:
I would not say so. Look at the std dev. its much higher ( reason could be random). I would they are wash.

Following are the results with 10k value size, full access slightly better, and sparse access is still order of magnitude faster as expected.

**without change:**
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess                                avgt    3  11084.838 ±  657.974   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess8                              avgt    3   4054.618 ±  387.857   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess16                             avgt    3   3576.869 ±  315.574   ns/op

**with change:**
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess                                avgt    3  10970.760 ±  326.789   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess8                              avgt    3   1290.028 ±  417.063   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess16                            avgt    3    885.955 ±  345.406   ns/op

Reply:
ok. can we add a public method to primitive arrays that makes sure they are loaded? that way users who care and know theyre about to access the entire array (maybe even several times over) can call it.

Reply:
Sure, we can add a getCached() type of public method to it whoever wants to iterate the full array multiple times. I increased the iteration to 5 and still see one full array access is not degraded at all. 

**without change:**
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess                                 avgt    5  11239.252 ± 548.146   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess8                               avgt    5   4140.900 ± 229.986   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess16                              avgt    5   3671.041 ± 269.680   ns/op

**with change:**
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess                                  avgt    5  11073.572 ± 109.950   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess8                                avgt    5   1314.574 ±  25.884   ns/op
FastAvroSerdesBenchmark.testFastAvroDeserializationAccess16                              avgt    5    917.803 ± 209.712   ns/op